File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#include <string.h>

#define MAX_STACK_SIZE 10000

int stack[MAX_STACK_SIZE];
int top = -1;

int isEmpty() {
    return (top < 0);
}

int push(int item) {
    if (top >= MAX_STACK_SIZE - 1) {
        return -1; // 스택이 가득 찼을 때는 -1 반환
    }
    stack[++top] = item;
    return item;
}

int pop() {
    if (isEmpty()) {
        return -1;
    }
    return stack[top--];
}

int peek() {
    if (isEmpty()) {
        return -1;
    }
    return stack[top];
}

int main() {
    int n;
    scanf("%d", &n);

    char command[10];
    int value;

    for (int i = 0; i < n; i++) {
        scanf("%s", command);

        if (strcmp(command, "push") == 0) {
            scanf("%d", &value);
            push(value);
        }
        else if (strcmp(command, "pop") == 0) {
            printf("%d\n", pop());
        }
        else if (strcmp(command, "size") == 0) {
            printf("%d\n", top + 1);
        }
        else if (strcmp(command, "empty") == 0) {
            printf("%d\n", isEmpty());
        }
        else if (strcmp(command, "top") == 0) {
            printf("%d\n", peek());
        }
    }

    return 0;
}

File: 백준/Silver/10828. 스택/스택.kt
import java.util.Stack

fun main() {
    val num = readln().toInt()
    val stack = Stack<String>()
    for(i in 0 until num){
        val str = readln()
        when (str[0]){
            'p' -> if(str[1] == 'u'){
                stack.push(str.split(" ")[1])
                }
                else if (stack.isEmpty()) println(-1) else println(stack.pop())
            't' -> println( if(stack.empty()) -1 else stack.peek() )
            's' -> println ( stack.size )
            'e' -> println ( if (stack.isEmpty() ) 1 else 0 )
            else -> println("뭔")
        }
    }
}